packages/@aws-cdk/aws-apigateway/lib/integrations/stepfunctions.ts

```diff
@@ -0,0 +1,206 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import { Token } from '@aws-cdk/core';
+import { ExecutionInput, ExecutionInputBuilder } from '.';
+import { IntegrationConfig, IntegrationOptions, PassthroughBehavior } from '../integration';
+import { Method } from '../method';
+import { AwsIntegration } from './aws';
+
+/**
+ * Options when configuring Step Functions integration with Rest API
+ */
+export interface StepFunctionsIntegrationOptions extends IntegrationOptions {
+  /**
+   * Check if cors is enabled
+   * @default false
+   */
+  readonly corsEnabled?: boolean;
+
+  /**
+   * You can add requestContext (similar to input requestContext from lambda input)
+   * to the input. The 'requestContext' parameter includes account ID, user identity, etc.
+   * that can be used by customers that want to know the identity of authorized users on
+   * the state machine side. The following code defines a REST API like above but also
+   * adds 'requestContext' to the input of the State Machine:
+   *
+   * @example
+   *
+   *    const stateMachine = new stepFunctions.StateMachine(this, 'StateMachine', ...);
+   *    new apigateway.StepFunctionsRestApi(this, 'StepFunctionsRestApi', {
+   *      stateMachine: stateMachine,
+   *      includeRequestContext: true,
+   *    });
+   *
+   * @default false
+   */
+  readonly includeRequestContext?: boolean;
+
+}
+/**
+ * Integrates a Synchronous Express State Machine from AWS Step Functions to an API Gateway method.
+ *
+ * @example
+ *
+ *    const stateMachine = new sfn.StateMachine(this, 'MyStateMachine', ...);
+ *    api.addMethod('GET', new StepFunctionsIntegration(stateMachine));
+ */
+export class StepFunctionsIntegration extends AwsIntegration {
+  private readonly stateMachine: sfn.IStateMachine;
+
+  constructor(stateMachine: sfn.IStateMachine, options: StepFunctionsIntegrationOptions = { }) {
+
+    const integResponse = integrationResponse();
+    const requestTemplate = requestTemplates(stateMachine, options.includeRequestContext);
+    super({
+      service: 'states',
+      action: 'StartSyncExecution',
```

Comment:
Looking at the docs here - https://docs.aws.amazon.com/step-functions/latest/dg/tutorial-api-gateway.html - looks like it's possible to call `StartExecution` as well. Why hardcode this here?

Reply:
Based on the comment above and the work below regarding integration response, looks like there is a layering problem here in the design.

`StepFunctionsIntegration` should be fully configurable by the user to do whatever is possible via CloudFormation in the context of configuring a state machine as the API Gateway backend.

It is acceptable `StepFunctionsRestApi` to be more opinionated than `StepFunctionsIntegration` by configuring specific integration response and forcing the use of express workflows.

So some of these opinionated choices need to move into `StepFunctionsRestApi` from here.